crates/astria-bridge-withdrawer/src/withdrawer/submitter/builder.rs

```diff
@@ -0,0 +1,99 @@
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+
+use astria_eyre::eyre::{
+    self,
+    ensure,
+    Context as _,
+};
+use sequencer_client::tendermint_rpc;
+use tokio_util::sync::CancellationToken;
+use tracing::{
+    info,
+    warn,
+};
+
+use super::state::State;
+
+const BATCH_QUEUE_SIZE: usize = 256;
+
+pub(crate) struct Builder {
+    pub(crate) shutdown_token: CancellationToken,
+    pub(crate) sequencer_key_path: String,
+    pub(crate) sequencer_chain_id: String,
+    pub(crate) cometbft_endpoint: String,
+    pub(crate) state: Arc<State>,
+}
+
+impl Builder {
+    /// Instantiates an `Submitter`.
+    pub(crate) async fn build(self) -> eyre::Result<(super::Submitter, super::Handle)> {
+        let Self {
+            shutdown_token,
+            sequencer_key_path,
+            sequencer_chain_id,
+            cometbft_endpoint,
+            state,
+        } = self;
+
+        let signer = super::signer::SequencerSigner::from_path(sequencer_key_path)?;
+        info!(address = %telemetry::display::hex(&signer.address), "loaded sequencer signer");
+        let (batches_tx, batches_rx) = tokio::sync::mpsc::channel(BATCH_QUEUE_SIZE);
+
+        let sequencer_cometbft_client = sequencer_client::HttpClient::new(&*cometbft_endpoint)
+            .context("failed constructing cometbft http client")?;
+
+        let actual_chain_id = get_sequencer_chain_id(sequencer_cometbft_client.clone()).await?;
```

Comment:
This could probably be moved into a "runtime intializer" that runs before the `select!` loop

This would remove the need to make this method async.